### ISSUE TITLE
feat: Add download as CSV to Data grid

### ISSUE THREAD

**Issue by Ogglas** [microsoft/fluentui-blazor#440]
# 🙋 Feature Request

It would be great with a CSV export feature for data grid.  See MUI as an example: https://mui.com/x/react-data-grid/export/

## 🤔 Expected Behavior

Button named export that exports headers and values into a CSV file.

**Comment by vnbaaij** [comment]
I don't think this is something we would make available out of the box. But I'll turn this into a discussion on the form so we can gather more feedback
